dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryException.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+
+namespace Microsoft.SemanticKernel.Connectors.Memory.AzureCognitiveSearch;
+
+/// <summary>
+/// Exception thrown by the Azure Cognitive Search connector
+/// </summary>
+public class AzureCognitiveSearchMemoryException : Exception
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureCognitiveSearchMemoryException"/> class.
+    /// </summary>
+    /// <param name="message">Exception message.</param>
+    public AzureCognitiveSearchMemoryException(string? message) : base(message)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AzureCognitiveSearchMemoryException"/> class.
+    /// </summary>
+    /// <param name="message">Exception message.</param>
+    /// <param name="innerException">Inner exception.</param>
+    public AzureCognitiveSearchMemoryException(string? message, Exception? innerException) : base(message, innerException)
+    {
+    }
+
+    private AzureCognitiveSearchMemoryException()
+    {
+        // Do not use, error message is required
+    }
```

Comment:
This should either be public or deleted.

Reply:
IIRC the build throws warnings if it's not defined. We make it private to stop people from raising exceptions without providing useful information.

Reply:
That's because there's an analyzer enabled that's checking that condition; its actual intent is to ensure that such ctors exist and are public. If weet don't want to expose such ctors, the analyzer should be disabled, or if we don't want to disable it everywhere, it should instead be suppressed here. Adding such a private ctor is an expebsive way to suppress the analyzer :-)

Reply:
solved